dex/order/order.go

```diff
@@ -473,6 +473,42 @@ func (t *Trade) SwapAddress() string {
 	return t.Address
 }
 
+// FromAccount is the account that the order originates from. Only useful for
+// account-based assets. Use of this method assumes that account coin has
+// already been added.
+func (t *Trade) FromAccount() string {
+	if len(t.Coins) == 0 {
+		return "no coins?"
+	}
+	return string(t.Coins[0])
```

Comment:
Shouldn't this be `t.Coins[0].String()`?

Reply:
> The account address should be utf-8 encoded

I've chosen to use the utf-8 encoding of the address hex string rather than the the address bytes so that we can do `string(coinID)` anywhere, regardless of whether we have a `msgjson.CoinID` or a `[]byte`. The alternative would require modifying field types in a number of places, if I remember correctly.

Reply:
OK, I see. So the client when setting this will basically do `.Coins[i] = []byte(addressStr)`?

Reply:
Actually, can you specify `account address should be utf-8 encoded`?  Is the address prior to casting to `[]byte` obtained from `common.Address.String()` or `common.Address.Hex()` or something else?

Reply:
Uh huh. That's what I was going for. I can doc this here.

Reply:
I think our messages crossed there, but here's what I'm looking at

```
// The coin ID should be the UTF-8 encoded address string, not the address
// byte-array. t.Coins[0] = []byte(addrStr).
```

Reply:
K, that's what I figured, but my question now is where does `addrStr` come from. https://github.com/decred/dcrdex/pull/1293#discussion_r760601794
The canonical encoding, hex with a 0x prefix, or just bare hex, lower case, upper, etc?

Reply:
I see. Yeah, we'll have to standardize that. Either of your suggestions would be fine. We could define a `dex/networks/eth` function, `func FundingCoinID(string) []byte`, if not just to enforce a standardization and provide documentation.

Reply:
The bulletproof approach I think would be to use the `common.Address[:]` bytes, and have `FromAccount` return bytes and the `AccountBalancer` take bytes too for the account to check.  I sense you thought through that but I'm not clear on the complications with that.